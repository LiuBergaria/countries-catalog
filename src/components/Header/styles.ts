import { StyleSheet } from 'react-native';

import Colors from 'src/styles/Colors';

const styles = StyleSheet.create({
  container: {
    padding: 16,
  },
  titleContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    position: 'relative',
  },
  title: {
    textAlign: 'center',
    color: Colors.white,
    fontWeight: 'bold',
    fontSize: 28,
  },
  subtitle: {
    textAlign: 'center',
    color: Colors.lightGrey,
    fontSize: 16,
  },
  loader: {
    position: 'absolute',
    left: 24,
  },
});

export default styles;
