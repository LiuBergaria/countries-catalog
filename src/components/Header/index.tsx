import React from 'react';
import { ActivityIndicator, Text, View } from 'react-native';

import Colors from 'src/styles/Colors';

import styles from './styles';

interface IProps {
  isLoading: boolean;
}

const Header = ({ isLoading }: IProps): JSX.Element => {
  return (
    <View style={styles.container}>
      <View style={styles.titleContainer}>
        <Text style={styles.title}>Countries List</Text>

        {isLoading && (
          <ActivityIndicator style={styles.loader} color={Colors.lightGrey} />
        )}
      </View>

      <Text style={styles.subtitle}>with infinite scroll</Text>
    </View>
  );
};

export default React.memo(Header);
