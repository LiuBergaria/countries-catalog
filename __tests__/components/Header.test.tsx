import React from 'react';
import renderer from 'react-test-renderer';
import Header from 'src/components/Header';

test('renders Header correctly', () => {
  const tree = renderer.create(<Header isLoading={false} />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('renders Header loading correctly', () => {
  const tree = renderer.create(<Header isLoading={true} />).toJSON();

  expect(tree).toMatchSnapshot();
});
