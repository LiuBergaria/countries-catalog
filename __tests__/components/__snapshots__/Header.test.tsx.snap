// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header correctly 1`] = `
<View
  style={
    Object {
      "padding": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "position": "relative",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 28,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Countries List
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#a7a7a7",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    with infinite scroll
  </Text>
</View>
`;

exports[`renders Header loading correctly 1`] = `
<View
  style={
    Object {
      "padding": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "position": "relative",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 28,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Countries List
    </Text>
    <ActivityIndicator
      color="#a7a7a7"
      style={
        Object {
          "left": 24,
          "position": "absolute",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#a7a7a7",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    with infinite scroll
  </Text>
</View>
`;
